my-code/significa/src/pages/Home.js

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect } from "react";
+import { Link } from "@reach/router";
+
+import Layout from "../components/layout";
+import MovieCard from "../components/MovieCard";
+import magnifier from "../assets/icons/icon-magnifier-grey.svg";
+import illustrationSm from "../assets/illustrations/illustration-empty-state.png";
+import illustrationLg from "../assets/illustrations/illustration-empty-state@2x.png";
+import "./Home.css";
+
+const API_KEY = `${process.env.REACT_APP_API_KEY}`;
+
+const Home = () => {
+  /* set variables that use state */
+  const [query, setQuery] = useState("");
+  const [empty, setEmpty] = useState(true);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [movies, setMovies] = useState([]);
+
+  /* function that handles change in the search bar */
+  const handleChange = (e) => {
+    /* console.log(e.target.value); */
+    if (e.target.value.trim() === "") {
+      setEmpty(true);
+    } else {
+      setEmpty(false);
+    }
+    return setQuery(e.target.value.trim());
+  };
+
+  /* fetch list of movies from the api when query changes */
+  useEffect(() => {
+    let mounted = true;
```

Comment:
I didn't get the reason for this, which kind of problem did you encountered here?